tests/parsers/test.manifestjson.js

```diff
@@ -332,17 +332,96 @@ describe('ManifestJSONParser', function() {
 
   describe('content security policy', function() {
 
-    it('should warn that csp will mean more review', () => {
+    it('dont warn on CSP not including default-src or script-src)', () => {
       var addonLinter = new Linter({_: ['bar']});
-      var json = validManifestJSON({content_security_policy: 'wat?'});
-      var manifestJSONParser = new ManifestJSONParser(json,
-                                                      addonLinter.collector);
+      var json = validManifestJSON({
+        content_security_policy: 'foo',
+      });
+      var manifestJSONParser = new ManifestJSONParser(
+        json, addonLinter.collector);
+
       expect(manifestJSONParser.isValid).toEqual(true);
-      var warnings = addonLinter.collector.warnings;
-      expect(warnings[0].code).toEqual(messages.MANIFEST_CSP.code);
-      expect(warnings[0].message).toContain('content_security_policy');
+      expect(addonLinter.collector.warnings.length).toEqual(0);
+    });
+
+    it('should warn on invalid values according to Add-On Policies', () => {
+      const invalidValues = [
+        'default-src *',
+        'default-src moz-extension: *',
+        'default-src ws:',
+        'default-src wss:',
+        'default-src http:',
+        'default-src https:',
+        'default-src ftp:',
+        'default-src http://cdn.example.com/my.js',
+        'default-src https://cdn.example.com/my.js',
+
+        'script-src *',
+        'script-src moz-extension: *',
+        'script-src ws:',
+        'script-src wss:',
+        'script-src http:',
+        'script-src https:',
+        'script-src ftp:',
+        'script-src http://cdn.example.com/my.js',
+        'script-src https://cdn.example.com/my.js',
+
+        // Properly match mixed with other directives
+        'script-src https: \'unsafe-eval\'; object-src \'self\'',
```

Comment:
Unfortunately not allowed by our eslint rules currently but a double-quote works too.